src/components/shared/memoBlocks/MemoBlocksGame.tsx

```diff
@@ -0,0 +1,172 @@
+'use client';
+
+import { closestCenter, DndContext } from '@dnd-kit/core';
+import {
+    arrayMove,
+    horizontalListSortingStrategy,
+    SortableContext,
+} from '@dnd-kit/sortable';
+import { useEffect, useState } from 'react';
+import { MatrixWithId } from '@/types';
+import SortableCard from './SortableCard';
+import { levelOneCards } from '@/mockData/memoBlocks';
+import { parseTable } from '@/lib/utils';
+import MemoBlocksCard from './MemoBlocksCard';
+import { FlipHorizontal2Icon, FlipVertical2Icon } from 'lucide-react';
+import { colorsTemplateMatrix } from '@/constants';
+
+
+type DragEventType = {
+    activatorEvent: PointerEvent;
+    active: {
+        id: string;
+        data: any;
+        rect: any;
+    };
+    collisions: any[];
+    delta: {
+        x: number;
+        y: number;
+        scaleX: number;
+        scaleY: number;
+    };
+    over: {
+        id: string;
+        rect: any;
+        data: any;
+        disabled: boolean;
+    };
+};
+
+function mirrorMatrix(matrix: MatrixWithId) {
+    if (!matrix) return;
+    const refMatrix = colorsTemplateMatrix;
+    const copyMatrix: MatrixWithId = { id: matrix.id, data: [] };
+    const rows = refMatrix.length;
+    const cols = refMatrix[0].length;
+    for (let i = 0; i < rows; i++) {
+        copyMatrix.data.push([]);
+        for (let j = 0; j < cols; j++) {
+            if (!matrix.data[rows - 1 - i][j].isActive)
+                copyMatrix.data[i].push({
+                    note: refMatrix[i][j].note,
+                    isActive: false,
+                    isTied: false,
+                });
+            else copyMatrix.data[i].push(refMatrix[i][j]);
+        }
+    }
+    return copyMatrix;
+}
+
+function flipMatrix(matrix: MatrixWithId) {
+    if (!matrix) return;
+    const copyMatrix: MatrixWithId = { id: matrix.id, data: [] };
+    const rows = matrix.data.length;
+    const cols = matrix.data[0].length;
+    for (let i = 0; i < rows; i++) {
+        copyMatrix.data.push([]);
+        for (let j = 0; j < cols; j++) {
+            copyMatrix.data[i].push(matrix.data[i][cols - 1 - j]);
+        }
+    }
+    return copyMatrix;
+}
+
+const setInitialMatrixes = () => {
+    function scrambleMatrix(matrix: MatrixWithId) {
+        const [flip, mirror] = [Math.random() > 0.5, Math.random() > 0.5];
+        let newMatrix = matrix;
+        if (flip) newMatrix = flipMatrix(newMatrix) as MatrixWithId;
+        if (mirror) newMatrix = mirrorMatrix(newMatrix) as MatrixWithId;
+        return newMatrix;
+    }
+
+    const levelCards = levelOneCards.map(table => parseTable(table));
+    const matrixes = levelCards.map(matrix => scrambleMatrix(matrix));
+    // switch order of matrixes
+    const matrixesCopy = [...matrixes];
+    matrixesCopy.sort(() => Math.random() - 0.5);
+    return matrixesCopy;
+};
+
+const MemoBlocksGame = () => {
```

Comment:
The convention in our project is to name the main component the same as its directory. In this case, the component should be named 'MemoBlocks,' or alternatively, consider changing the directory name to 'memoBlocksGame